cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityManager.java

```diff
@@ -0,0 +1,147 @@
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Class with helpful methods for managing capabilities
+ */
+public class CapabilityManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CapabilityManager.class);
+  private static final String CAPABILITY = "capability:%s";
+  private static final String APPLICATION_TAG = "application:%s";
+  private static final String CAPABILITY_KEY = "capability";
+  private static final String APPLICATION = "application";
+  private final MetadataSearchClient metadataClient;
+  private final CConfiguration cConf;
+
+  @Inject
+  CapabilityManager(MetadataSearchClient metadataClient, CConfiguration cConf) {
+    this.metadataClient = metadataClient;
+    this.cConf = cConf;
+  }
+
+  /**
+   * Returns the list of applications that are having metadata tagged with the capability
+   *
+   * @param namespace  Namespace for which applications should be listed
+   * @param capability Capability by which to filter
+   * @param cursor Optional cursor from a previous response
+   * @param offset Offset from where to start
+   * @param limit Limit of records to fetch
+   * @return
+   * @throws Exception - Exception from meta data search if any
+   */
+  public CapabilityApplications getAppsForCapability(NamespaceId namespace, String capability, @Nullable String cursor,
+                                                     int offset, int limit) throws IOException {
+    String capabilityTag = String.format(CAPABILITY, capability);
+    SearchRequest searchRequest = SearchRequest.of(capabilityTag)
+      .addNamespace(namespace.getNamespace())
+      .addType(APPLICATION)
+      .setScope(MetadataScope.SYSTEM)
+      .setCursor(cursor)
+      .setOffset(offset)
+      .setLimit(limit)
+      .build();
+    MetadataSearchResponse searchResponse = metadataClient.search(searchRequest);
+    Set<ApplicationId> applicationIds = searchResponse.getResults().stream()
+      .map(MetadataSearchResultRecord::getMetadataEntity)
+      .map(this::getApplicationId)
+      .collect(Collectors.toSet());
+    return new CapabilityApplications(applicationIds, getCursorResponse(searchResponse), searchResponse.getOffset(),
+                                      searchResponse.getLimit(), searchResponse.getTotal());
+  }
+
+  @Nullable
+  private String getCursorResponse(MetadataSearchResponse searchResponse) {
+    List<String> cursors = searchResponse.getCursors();
+    if (cursors == null || cursors.isEmpty()) {
+      return null;
+    }
+    return cursors.get(0);
+  }
+
+  /**
+   * Returns boolean indicating whether application is disabled due to a disabled capability
+   *
+   * @param namespace
+   * @param capabilityName
+   * @return
+   * @throws Exception
+   */
+  public boolean isApplicationDisabled(String namespace, String capabilityName) throws IOException {
```

Comment:
Use positive method name `isApplicationEnabled` and perform positivity check instead.

Reply:
Changed